noxfile.py

```diff
@@ -146,6 +153,7 @@ def test_cartesian(
         "*.toml",
         "*.yml",
         "noxfile*.py",
+        ".python-versions",
```

Comment:
Either we add this to all sessions or to none of them, but only add it to `cartesian` doesn't make sense to me. Personally, I wouldn't add anything since if we change the supported python versions we are also changing the `pyproject.toml` (and most likely `noxfile.py` too, so it shouldn't be needed)